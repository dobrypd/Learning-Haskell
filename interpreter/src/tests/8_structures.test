struct {
    int a = 10;
    int b = 20;
} first;
struct {
    int a = 30;
    int b = 40;
} second;

int main() {
    return first.a + first.b + second.a * second.b;
}
