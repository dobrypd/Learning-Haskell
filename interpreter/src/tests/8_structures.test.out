Welcome to Kappa interpreter
1230
[(fromList [(Ident "first",fromList [(Ident "a",10),(Ident "b",20)]),(Ident "second",fromList [(Ident "a",30),(Ident "b",40)])],fromList [(Ident "main",([],[SStm (JumpS (SjumpFive (Eplus (Eplus (Evar [Ident "first",Ident "a"]) (Evar [Ident "first",Ident "b"])) (Etimes (Evar [Ident "second",Ident "a"]) (Evar [Ident "second",Ident "b"])))))]))],fromList [])]
